### ISSUE TITLE
Permit Colab links in addition to or instead of Binder links

### ISSUE THREAD

**Issue by nathancarter** [executablebooks/sphinx-book-theme#107]
It is nice and easy to turn any Jupyter notebook on GitHub into an executable one on Colab by simply visiting any URL of the form `https://colab.research.google.com/github/USER/REPO/blob/master/PATH/FILENAME.ipynb`.  This is often more desirable than Binder because of the much faster load time of the Colab environment than the Binder one, and of course the ability to save your own copies of notebooks.  Could this be an option in jupyter-book, at least for those books published to GitHub?

**Comment by nathancarter** [comment]
My apologies, but I am having a busier summer than I thought, and I will not have the bandwidth to join in executablebooks development any time soon.

**Comment by lewtun** [comment]
i'm happy to have a stab at this. will report back if i get stuck.

**Comment by lewtun** [comment]
hi @choldgraf, i've implemented a minimal version of the colab button [here](https://github.com/lewtun/sphinx-book-theme/tree/feature/add-colab-launch-button), but before i make a pr i have a few questions:

1. i could not find an svg colab icon, so have resorted to png. is this ok?
2. as far as i can tell, colab can't open markdown files which suggests i should remove colab from the dropdown in `docs/notebooks.md`.
![Screen Shot 2020-06-17 at 11 41 07 am](https://user-images.githubusercontent.com/26859204/84882439-8bc55a00-b08f-11ea-9d62-061fabebe0bd.png)
3. if point (2) is correct, what would be the best way to test the button is launching correctly? (one idea is to create an editable install of jupyter book with my fork of sbt, but that seems a bit cumbersome)

**Comment by choldgraf** [comment]
nice! looking really good :-)

a PNG is fine with me, but could you find one that has a transparent background? (and minify it so that it is as tiny as possible)

for markdown pages, yeah that works to just not show colab on those pages.

how do you mean "test the button launching correctly"? What I've done is just looked at the URL link structure that is generated for Binder, and confirm that it is correct (e.g.: https://github.com/executablebooks/sphinx-book-theme/blob/master/tests/test_build.py#L54). This may not be implemented for launch buttons yet, but if you're willing to give it a shot it would be a nice time to do so :-)

You can see how we test for this with the edit buttons here: https://github.com/executablebooks/sphinx-book-theme/blob/master/tests/test_build.py#L194

Basically, you'd need to convert the manual checks for binderhub etc buttons here: https://github.com/executablebooks/sphinx-book-theme/blob/master/tests/test_build.py#L54 into regressions tests similar to the link I shared above, and then add the collaboratory to that as well.

does that make sense?

**Comment by lewtun** [comment]
thanks for the feedback and advice @choldgraf!

i've made the colab logo transparent and it's now of similar size to the jupyterhub one:
![Screen Shot 2020-06-18 at 11 38 42 am](https://user-images.githubusercontent.com/26859204/85005116-a8799480-b158-11ea-9bcd-857f54e13875.png)

i also included a url link test for the colab button (https://github.com/lewtun/sphinx-book-theme/blob/feature/add-colab-launch-button/tests/test_build.py#L62) but am finding that the button is not appearing in the html, despite the fact that i include it in the `conf.py` test file: https://github.com/lewtun/sphinx-book-theme/blob/feature/add-colab-launch-button/tests/sites/base/conf.py#L28

by "not appearing" i mean that with a debugger i can open the html produced by `sphinx_buld.build()` and visually see that only the binder / jupyterhub / thebelab buttons are present:

![Screen Shot 2020-06-18 at 11 46 10 am](https://user-images.githubusercontent.com/26859204/85005587-508f5d80-b159-11ea-99e6-5cb4de79ef48.png)

this obviously throws an exception in the unit test, but i am quite mystified why this is the case. 

what is unusual is that with a debugger i can pause the test before `sphinx_build.build()` is called, manually drill into the test folder created by `sphinx_build.copy()` and create the html by running
```
sphinx-build . _build/html -a -W
```
in the terminal. in this case the colab button appears 🤷‍♂️ . do you happen to know if i am missing some obvious config for the test?

**Comment by choldgraf** [comment]
hmmm - can you make a pull request and we can start to iterate there? then we can also see the failing test

**Comment by lewtun** [comment]
sure - [done](https://github.com/executablebooks/sphinx-book-theme/pull/123)!

for reasons that i don't understand, adding the file regression test appears to have fixed the pytest problems i was running into above. probably not worth digging into since i am not able to reproduce it - sorry for the noise!
